### DISCUSSION TITLE
OrderBy _sum of relation's field

### DISCUSSION THREAD

**Discussion by VaibhavAcharya** [prisma/prisma#15780] [Q&A] [answered]
# Goal
Order `institutes` by `_sum` of `rating` which are part of `reviews` realtion.

# Schema 
```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = []
}

datasource db {
  provider = "cockroachdb"
  url      = env("DATABASE_URL")
}

model Institute {
  id   String @id @default(cuid())
  slug String @unique

  name       String
  websiteURL String

  countryID String
  country   Country @relation(fields: [countryID], references: [id], onUpdate: Cascade, onDelete: Cascade)

  reviews Review[]
  views   Int      @default(0)
}

model Country {
  id String @id @default(cuid())

  code String @unique
  name String @unique

  institutes Institute[]
}

model Review {
  id String @id @default(cuid())

  instituteID String
  institute   Institute @relation(fields: [instituteID], references: [id], onUpdate: Cascade, onDelete: Cascade)

  rating  Int
  content String?

  @@unique([instituteID, email])
}
```

# Problem
```js
const institutes = await db.institute.findMany({
  orderBy: {
    reviews: {
      _count: true,
      // No other option available! 😢
    },
    country: {
      code: "desc",
      // I do get all options here though. 🤔
    }
  }
});
```

**Comment by nurul3101** [comment, marked as answer]
Hello @VaibhavAcharya 👋 

orderByRelation has support for `_count` as of now, we have a similar feature request to add support for other filters as well: [#8190](https://github.com/prisma/prisma/issues/8190)
You could use [Raw Query](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access) as a workaround for now.
Also on a side note, it seems you have defined email in the unique index, but email is not a part of the review model.

**Reply by VaibhavAcharya** [reply]
Thanks!

Regarding email, I actually filtered out unrelated tables and fields for this schema.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
